### PR TITLE
Let thread func have optional parameter

### LINKED ISSUE #38042: Threads dont work with optional arguments
**Godot version:**
v3.2.1

**OS/device including version:**
Windows 10

**Issue description:**
Starting a new thread on a function with optional arguments doesn't work as expected.
Both calling the function normally and calling the function by emitting a signal works perfectly as expected.

**Minimal reproduction project:**
```gdscript
extends Node2D

var thread : Thread

signal test_signal

func _ready():
	thread = Thread.new()
	thread.start(self, "do_stuff")

	connect("test_signal", self, "do_stuff")
	emit_signal("test_signal")
	
	do_stuff()

func do_stuff(optional_var : int = 0):
	print("HURRAY!")

func _exit_tree():
	thread.wait_to_finish()
```

Gives following error when running:
```
E 0:00:00.812   _start_func: Could not call function 'do_stuff' to start thread 13480: Invalid Argument #0.
  <C++ Error>   Method failed.
  <C++ Source>  core/bind/core_bind.cpp:2694 @ _start_func()
```

Evidently running the start thread command with an actual integer as argument does work.

### PR DESCRIPTION
**Pull request by Faymoon**
Fixes [#38042](https://github.com/godotengine/godot/issues/38042)

### CODE REVIEW

**Review by Calinou — APPROVED** [review]
Looks good to me, but this should be tested on the latest `master` and `3.x` branches before it can be merged (with the [multi-threading modernization](https://github.com/godotengine/godot/pull/45618) in place).

File: core/bind/core_bind.cpp [hunk 1]
```diff
@@ -2098,10 +2098,11 @@ void _Thread::_start_func(void *ud) {
 	memdelete(tud);
 	Callable::CallError ce;
 	const Variant *arg[1] = { &t->userdata };
+	int argc = arg[0]->get_type() != 0;
```
Comment (akien-mga) [inline core/bind/core_bind.cpp:2101]: This should use `Variant::NIL`, not `0`.

And I'd make the cast from `bool` to `int` explicit:

```suggestion
	int argc = (int)(arg[0]->get_type() != Variant::NIL);
```

**Comment by akien-mga** [comment]
Sorry for the late review.

This would need a rebase for the `master` branch (`core/bind/core_bind.cpp` moved to `core/core_bind.cpp`) but otherwise the code is still the same so this fix might still be relevant.

The fix looks OK to me but I'll let @RandomShaper or @reduz validate as they know this code better.

FYI, the email used to author your commit (see https://patch-diff.githubusercontent.com/raw/godotengine/godot/pull/38078.patch) is not affiliated to your GitHub account. It's not a problem for Git, but if you want to claim ownership of that commit, you can add this email as secondary email in your GH settings.

**Review by RandomShaper — APPROVED** [review]

**Comment by akien-mga** [comment]
Force pushed as a rebase that solves merge conflict and my above comment.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**